llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -58164,6 +58164,16 @@ static SDValue combineX86AddSub(SDNode *N, SelectionDAG &DAG,
     }
   }
 
+  // Fold ADD(ADC(Y, C1, CF), C2) -> ADC(Y, C1 + C2, CF)
+  if (!IsSub && LHS.getOpcode() == X86ISD::ADC &&
+      isa<ConstantSDNode>(LHS.getOperand(1)) && isa<ConstantSDNode>(RHS)) {
+    auto *C1 = dyn_cast<ConstantSDNode>(LHS.getOperand(1));
+    auto *C2 = dyn_cast<ConstantSDNode>(RHS);
+    APInt Sum = C1->getAPIntValue() + C2->getAPIntValue();
+    return DAG.getNode(X86ISD::ADC, DL, N->getVTList(), LHS.getOperand(0),
```

Comment:
This combine can produce different results for the O and C flags. I think we need to know that only Z or S flag is used. Is that already checked somewhere else?